frontend/src/pages/DiscoveryHook.tsx

```
@@ -56,7 +56,13 @@ const DiscoveryHook: React.FC = () => {
       try {
         setIsLoading(true);
         const data = await getMessages();
-        setMessages(data.messages || []);
+        // Normalize incoming messages to the shape expected by PreviewModal
+        const normalized = (data.messages || []).map((msg: any, idx: number) => ({
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)